zqd/handlers_test.go

```diff
@@ -576,7 +605,10 @@ func TestDeleteDuringPacketPost(t *testing.T) {
 	require.Error(t, <-packetPostErr, "context canceled")
 }
 
-func zngSearch(t *testing.T, client *api.Connection, space, prog string) string {
+// zngSearch runs the provided zql program as a search on the provided
+// space, returning a slice of control messages along with a strong
+// containing the tzng results.
+func zngSearch(t *testing.T, client *api.Connection, space, prog string) ([]interface{}, string) {
```

Comment:
Rather than add this extra return argument that non of the other tests use, I would suggest creating a different func for retrieving stats *or* just calling the api Connection directly in the one test.

Reply:
I think it's reasonable that a test helper running a search would return the control messages alongside the search results, even if most tests use just one or the other. Also, I think we have an indexing test in our near future that need to look at both. That's the context, but if you prefer I can make this suggested change.

Reply:
fwiw, I'm also ok with zngSearch returning both the control & results in one call.

Reply:
I would just prefer that you use `api.Connection` directly. No longer can I do things like:
```
require.Equal(t, zngSearch(t, "test", "* | count"), expected)
```

Reply:
Antidotally in the invalid the SearchRequest test will call the Connection directly. I think this is the right move, but why shouldn't we also now return `error` so we can use the same func?

Reply:
I would prefer that you just make a new function `search` that is the new `zngSearch` and make `zngSearch` do:
```
func zngSearch(args) string {
   str, _ := search(args)
   return str
}
```

Reply:
I don't have strong feelings one way or the other, but I do think a function returning both a tzng string and a []interface{} of control messages should return them as `string, []interface{}` since the string is almost always more interesting.

Reply:
Having thought about it a little more, I'm now of the opinion that two functions (maybe search, returning two values, and searchTzng, returning one) would be nice.

Reply:
Thanks for all the thoughts... updated to the suggestion in the most recent comment.

> Antidotally in the invalid the SearchRequest test will call the Connection directly. I think this is the right move, but why shouldn't we also now return error so we can use the same func?

@mattnibs I couldn't parse the above question... not sure if it still needs to be answered.